### PR TITLE
Backup list fmt

### PR DESCRIPTION
**Pull request by MiaAltieri**
Updating the formatting of `list-backups` to be of the agreed upon format:
```
      backup-id             | backup-type  | backup-status
      ----------------------------------------------------
      2023-02-08T15:19:34Z  | logical      | finished
      2023-02-09T10:13:46Z  | logical      | finished
      2023-02-14T13:54:48Z  | logical      | failed
      2023-02-14T13:58:27Z  | logical      | in progress
```
Added necessary unit tests as well

### CODE REVIEW

**Review by delgod — APPROVED** [review]

File: tests/unit/test_mongodb_backups.py [hunk 1]
```diff
@@ -530,3 +530,52 @@ def test_s3_credentials_pbm_error(self, output, defer, resync, _set_config_optio
...
+            output_contents = "".join(output_contents)
+
+        check_output.return_value = output_contents.encode("utf-8")
+        formatted_output = self.harness.charm.backups._generate_backup_list_output()
+        formatted_output = formatted_output.split("\n")
+        header = formatted_output[0]
+        self.assertEqual(header, "backup-id             | backup-type  | backup-status")
+        divider = formatted_output[1]
+        self.assertEqual(
+            divider, "-" * len("backup-id             | backup-type  | backup-status")
```
Comment (marceloneppel) [inline tests/unit/test_mongodb_backups.py:553]: Optional:
```suggestion
            divider, "-" * len(header)
```

File: tests/integration/backup_tests/test_backups.py [hunk 2]
```diff
@@ -133,8 +133,9 @@ async def test_create_and_list_backups(ops_test: OpsTest) -> None:
     # verify backup list works
     action = await db_unit.run_action(action_name="list-backups")
     list_result = await action.wait()
+    print(list_result)
```
Comment (marceloneppel) [inline tests/integration/backup_tests/test_backups.py:136]: Optional: change to `logger.info`.

File: lib/charms/mongodb/v0/mongodb_backups.py [hunk 3]
```diff
@@ -318,12 +319,61 @@ def _on_list_backups_action(self, event) -> None:
...
+        running_backup = pbm_status["running"]
+        if running_backup.get("type", None) == "backup":
+            # backups are sorted in reverse order
+            last_reported_backup = backup_list[0]
+            # pbm will occasionally report backups that are currently running as failed, so it is
+            # necessary to correct the backup list in this case.
+            if last_reported_backup[0] == running_backup["name"]:
+                backup_list[0] = (last_reported_backup[0], last_reported_backup[1], "in progress")
+            else:
+                backup_list.append((running_backup["name"], "logical", "in progress"))
```
Comment (marceloneppel) [inline lib/charms/mongodb/v0/mongodb_backups.py:360]: Is the backup type always logical in this case?

**Review by marceloneppel — APPROVED** [review]
Looks great!

File: lib/charms/mongodb/v0/mongodb_backups.py (hunk 3 above)
Comment (MiaAltieri) [reply lib/charms/mongodb/v0/mongodb_backups.py:360]: Yes, until we support physical backups any backup running on PBM will be logical :)

**Pull request merged**